tests/lib/rules/no-implicit-coercion.js

```diff
@@ -64,6 +64,10 @@ ruleTester.run("no-implicit-coercion", rule, {
         {code: "0 + foo"},
         {code: "~foo.bar()"},
 
+        // https://github.com/eslint/eslint/issues/7057
+        {code: "'' + 'foo'"},
+        {code: "+42"},
```

Comment:
Could you add a valid test for `"str += "non-empty"`? I didn't see it in the test file but you did change `isAppendEmptyString`. Thanks!